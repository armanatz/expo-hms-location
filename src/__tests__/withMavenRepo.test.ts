import {
  _addHuaweiMavenRepoToBuildScript,
  _addHuaweiMavenRepoToAllProjects,
  _addHMSAGConnectServicesPlugin,
  _pinAndroidGradleBuildToolsVersion,
} from '../actions/withMavenRepo';

const projectBuildGradle = `buildscript {
    repositories {
        google()
        jcenter()
        mavenCentral()
    }
    dependencies {
        classpath('com.android.tools.build:gradle')
    }
}
`;

describe('project build.gradle modifications', () => {
  it('adds huawei maven repo to buildscript', () => {
    // Test that it adds the repo correctly
    const firstResult = _addHuaweiMavenRepoToBuildScript(projectBuildGradle);
    expect(firstResult.contents).toMatchSnapshot();
    expect(firstResult.didInsertOrReplace).toBe(true);
    expect(firstResult.didClear).toBe(false);

    // Test that it doesn't duplicate the repo
    const secondResult = _addHuaweiMavenRepoToBuildScript(firstResult.contents);
    expect(secondResult.contents).toBe(firstResult.contents);
    expect(secondResult.didInsertOrReplace).toBe(false);
    expect(secondResult.didClear).toBe(false);
  });

  it('creates new allprojects and adds huaewi maven repo to it', () => {
    // Test that it adds the repo correctly
    const firstResult = _addHuaweiMavenRepoToAllProjects(projectBuildGradle);
    expect(firstResult.contents).toMatchSnapshot();
    expect(firstResult.didAppend).toBe(true);
    expect(firstResult.didClear).toBe(false);

    // Test that it doesn't duplicate the repo
    const secondResult = _addHuaweiMavenRepoToAllProjects(firstResult.contents);
    expect(secondResult.contents).toBe(firstResult.contents);
    expect(secondResult.didAppend).toBe(false);
    expect(secondResult.didClear).toBe(false);
  });

  it('adds agcp plugin to buildscript dependencies', () => {
    // Test that it adds the plugin correctly
    const firstResult = _addHMSAGConnectServicesPlugin(projectBuildGradle);
    expect(firstResult.contents).toMatchSnapshot();
    expect(firstResult.didMerge).toBe(true);
    expect(firstResult.didClear).toBe(false);

    // Test that it doesn't duplicate the plugin
    const secondResult = _addHMSAGConnectServicesPlugin(firstResult.contents);
    expect(secondResult.contents).toBe(firstResult.contents);
    expect(secondResult.didMerge).toBe(false);
    expect(secondResult.didClear).toBe(false);
  });

  it('pins android gradle build tools version', () => {
    // Test that it adds the version correctly
    const firstResult = _pinAndroidGradleBuildToolsVersion(projectBuildGradle);
    expect(firstResult.contents).toMatchSnapshot();
    expect(firstResult.didInsertOrReplace).toBe(true);
    expect(firstResult.didClear).toBe(false);

    // Test that it doesn't duplicate the version
    const secondResult = _pinAndroidGradleBuildToolsVersion(
      firstResult.contents,
    );
    expect(secondResult.contents).toBe(firstResult.contents);
    expect(secondResult.didInsertOrReplace).toBe(false);
    expect(secondResult.didClear).toBe(false);
  });
});
