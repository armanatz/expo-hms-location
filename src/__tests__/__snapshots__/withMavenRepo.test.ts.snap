// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project build.gradle modifications adds agcp plugin to buildscript dependencies 1`] = `
"buildscript {
    repositories {
        google()
        jcenter()
        mavenCentral()
    }
    dependencies {
        // @generated begin @armanatz/expo-hms-location-agconnect - expo prebuild (DO NOT MODIFY) sync-854ce4f5ef964a021eefccc1c88cdde6e9fad117
        classpath('com.huawei.agconnect:agcp:1.9.1.301')
        // @generated end @armanatz/expo-hms-location-agconnect
        classpath('com.android.tools.build:gradle')
    }
}
"
`;

exports[`project build.gradle modifications adds huawei maven repo to buildscript 1`] = `
"buildscript {
    repositories {
        google()
        jcenter()
        mavenCentral()
        // @generated begin @armanatz/expo-hms-location-maven-buildscript - expo prebuild (DO NOT MODIFY) sync-b660f0736d1147334376fd8aa31eb80c855d4be0
        maven { url 'https://developer.huawei.com/repo/' }
        // @generated end @armanatz/expo-hms-location-maven-buildscript
    }
    dependencies {
        classpath('com.android.tools.build:gradle')
    }
}
"
`;

exports[`project build.gradle modifications creates new allprojects and adds huaewi maven repo to it 1`] = `
"buildscript {
    repositories {
        google()
        jcenter()
        mavenCentral()
    }
    dependencies {
        classpath('com.android.tools.build:gradle')
    }
}
// @generated begin @armanatz/expo-hms-location-maven-allprojects - expo prebuild (DO NOT MODIFY) sync-87a1e6ffa6f51bf342280163ff2b9ae2d4ed71d6
allprojects {
    repositories {
        maven { url 'https://developer.huawei.com/repo/' }
    }
}
// @generated end @armanatz/expo-hms-location-maven-allprojects"
`;

exports[`project build.gradle modifications pins android gradle build tools version 1`] = `
"buildscript {
    repositories {
        google()
        jcenter()
        mavenCentral()
    }
    dependencies {
        // @generated begin @armanatz/expo-hms-location-buildscript-buildtools - expo prebuild (DO NOT MODIFY) sync-e1eab30c134a714f0755d42f4bb3bbe976d48d57
        classpath('com.android.tools.build:gradle:8.6.0')
        // @generated end @armanatz/expo-hms-location-buildscript-buildtools
    }
}
"
`;
